src/app/core/services/authorizationService/authorization.service.ts

```diff
@@ -1,48 +1,104 @@
-import { Injectable } from '@angular/core';
-import { Observable, Subject, ReplaySubject } from 'rxjs';
+import { Injectable, ChangeDetectionStrategy } from '@angular/core';
+import { Observable, Subject, ReplaySubject, Subscription,BehaviorSubject } from 'rxjs';
+import { Response, Request, RequestOptions, RequestMethod, Http, Headers, URLSearchParams } from '@angular/http';
+import { HttpService } from './../httpService/http.service';
+
 
 import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/switchMap';
+import 'rxjs/add/operator/catch';
 
 const GUEST_NAME = 'Guest';
 
 
 @Injectable()
 export class AuthorizationService {
 	private userInfo:{};
-	private autificated: boolean;
+	private autificated: BehaviorSubject<any>;
+
 	public userLogin$: Observable<string>;
+	public autificated$: Observable<boolean>;
+
 	private userLoginChange: ReplaySubject<string>;
+	private autificatedChange: ReplaySubject<boolean>;
+
+	private authLoginUrl: string = 'http://127.0.0.1:3004/auth/login';
+	private authUserInfoUrl: string = 'http://127.0.0.1:3004/auth/userinfo';
 
 
-	constructor() {
+	constructor( private http: HttpService) {
 		this.userInfo = {
-			id: 585507,
+			/*id: 585507,
 			login: 'Smurfik',
 			admin: true,
-			token: 'vm58mv45vm599,y54-v45w9vc9-vryc,o5y'
+			token: 'vm58mv45vm599,y54-v45w9vc9-vryc,o5y'*/
 		};
-		this.userLoginChange = new ReplaySubject<string>();
-		this.userLogin$ = this.userLoginChange.asObservable().startWith(this.userInfo['login']);
-		this.autificated = true;
+		this.userLoginChange = new ReplaySubject();
+		this.userLogin$ = this.userLoginChange.asObservable().startWith('');
+		this.autificatedChange	= new ReplaySubject();
+		this.autificated$ = this.autificatedChange.asObservable().startWith(false);
+		this.autificated = new BehaviorSubject(false);
+
+
 	}
 
-	public loginUser() {
-		this.autificated = true;
-		this.userLoginChange.next(this.userInfo['login']);
-		localStorage.setItem(this.userInfo['login'], this.userInfo['token']);
+	public loginUser(credentials: any) {
+		credentials = {
+			login:'Warner',
+			password:'ea'
+		};
+		this.userInfo ={
+			login:'Warner',
+			password:'ea'
+		}
+
+		let headers = new Headers({
+			'Accept': 'application/json'
+		});
+		headers.append('Content-Type', 'text/plain')
+		let options = new RequestOptions({ headers: headers });
```

Comment:
Just FYI, in this case we can use object shorthand notation
```javascript
let options = new RequestOptions({ headers });
```
Also, it's good approach to use `const` with variables which you are not going to change. IDE will help you to avoid rewriting value of this variable.